internal/transform/trade.go

```diff
@@ -0,0 +1,210 @@
+	if operationIndex >= int32(len(operationResults)) {
+		err = fmt.Errorf("Operation index of %d is out of bounds in result slice (len = %d)", operationIndex, len(operationResults))
+		return
+	}
+
+	if operationResults[operationIndex].Tr == nil {
+		err = fmt.Errorf("Could not get result Tr for operation at index %d", operationIndex)
+		return
+	}
+
+	operationTr, ok := operationResults[operationIndex].GetTr()
+	if !ok {
+		err = fmt.Errorf("Could not get result Tr for operation at index %d", operationIndex)
+		return
+	}
+
+	switch operationType {
+	case xdr.OperationTypeManageBuyOffer:
+		offerResult, ok := operationTr.GetManageBuyOfferResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
+			return
+		}
+
+		success, ok := offerResult.GetSuccess()
+		if !ok {
+			err = fmt.Errorf("Could not get %sSuccess for operation at index %d", operationType, operationIndex)
+			return
+		}
+
+		claimedOffers = success.OffersClaimed
+	case xdr.OperationTypeManageSellOffer:
+		offerResult, ok := operationTr.GetManageSellOfferResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		success, ok := offerResult.GetSuccess()
+		if !ok {
+			err = fmt.Errorf("Could not get %sSuccess for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		claimedOffers = success.OffersClaimed
+	case xdr.OperationTypeCreatePassiveSellOffer:
+		offerResult, ok := operationTr.GetCreatePassiveSellOfferResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		success, ok := offerResult.GetSuccess()
+		if !ok {
+			err = fmt.Errorf("Could not get %sSuccess for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		claimedOffers = success.OffersClaimed
+	case xdr.OperationTypePathPaymentStrictSend:
+		offerResult, ok := operationTr.GetPathPaymentStrictSendResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		success, ok := offerResult.GetSuccess()
+		if !ok {
+			err = fmt.Errorf("Could not get %sSuccess for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		claimedOffers = success.Offers
+	case xdr.OperationTypePathPaymentStrictReceive:
+		offerResult, ok := operationTr.GetPathPaymentStrictReceiveResult()
+		if !ok {
+			err = fmt.Errorf("Could not get %sResult for operation at index %d", operationType, operationIndex)
+			return
+
+		}
+
+		success, ok := offerResult.GetSuccess()
+		if !ok {
+			err = fmt.Errorf("Could not get %sSuccess for operation at index %d", operationType, operationIndex)
+			return
+		}
+
+		claimedOffers = success.Offers
```

Comment:
I think you can move this assignment to just before the final `return` (outside the `switch` statement). That way, you don't have to call it in every case.

Reply:
The success variable has a different type in every case, and sometimes the field names are different. I'm not sure how I would cleanly move this statement outside the switch.

Reply:
Got it, this makes sense on another read. Should be fine as-is.